pkg/build/builder/common.go

```diff
@@ -180,3 +155,47 @@ func execPostCommitHook(client DockerClient, postCommitSpec api.BuildPostCommitS
 		Stderr:       true,
 	})
 }
+
+func updateBuildRevision(build *api.Build, sourceInfo *git.SourceInfo) *api.SourceRevision {
+	if build.Spec.Revision != nil {
+		return build.Spec.Revision
+	}
+	return &api.SourceRevision{
+		Git: &api.GitSourceRevision{
+			Commit:  sourceInfo.CommitID,
+			Message: sourceInfo.Message,
+			Author: api.SourceControlUser{
+				Name:  sourceInfo.AuthorName,
+				Email: sourceInfo.AuthorEmail,
+			},
+			Committer: api.SourceControlUser{
+				Name:  sourceInfo.CommitterName,
+				Email: sourceInfo.CommitterEmail,
+			},
+		},
+	}
+}
+
+func updateStatusRetrier(build *api.Build, client client.BuildInterface, sourceRev *api.SourceRevision) error {
+	return kclient.RetryOnConflict(kclient.DefaultBackoff, func() error {
```

Comment:
Sounds like a broken telephone. We added Patch in the idling controller for DCs and RCs. I think Patch in controllers makes sense in cases where the resource to be updated is not the main resource owned by the controller and you always want the update to succeed (eg. you mark a deployment as {inert,cancelled,complete,...} or you adopt a replica set). That being said it doesn't seem that you use `retryBuildStatusUpdate` in any build controller but in the builder images?